src/main/scala/com/nec/spark/planning/OneStageEvaluationPlan.scala

```diff
@@ -50,6 +53,16 @@ final case class OneStageEvaluationPlan(
 
   override def output: Seq[Attribute] = outputExpressions.map(_.toAttribute)
 
+  private val isProject = veFunction.functionName.startsWith("project")
```

Comment:
I'd create a new plan with `Project` in its name.

Reply:
Done.